sparse/src/KokkosSparse_spiluk_handle.hpp

```diff
@@ -132,21 +132,26 @@ class SPILUKHandle {
         vector_size(-1) {}
 
   void reset_handle(const size_type nrows_, const size_type nnzL_,
-                    const size_type nnzU_, const size_type block_size_) {
+                    const size_type nnzU_, const size_type block_size_ = -1) {
     set_nrows(nrows_);
     set_num_levels(0);
     set_nnzL(nnzL_);
     set_nnzU(nnzU_);
-    set_block_size(block_size_);
+    // user likely does not want to reset block size to 0, so set default
+    // to -1
+    if (block_size_ != static_cast<size_type>(-1)) {
```

Comment:
@jgfouca can `size_type` be unsigned, and if so would this result in some unexpected wrap-around value?

Reply:
@ndellingwood , yes, it will be a huge value, something no one would ever pick for a block size.

Reply:
In that case I think it would be a bit more explicit to use `Kokkos::ArithTraits<size_type>::max()`. This would allow us to make comparisons more confidently down the road in my opinion.